### PR TITLE
French Translation - JavaScript Chapter

### PR DESCRIPTION
**Pull request by borisschapira**
Makes progress on #539

### CODE REVIEW

File: src/content/fr/2019/javascript.md [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+
+<figure>
+   <a href="/static/images/2019/javascript/js-processing-reddit.png">
+      <img src="/static/images/2019/javascript/js-processing-reddit.png" alt="Délais de traitement de JavaScript pour Reddit.com." aria-labelledby="fig4-caption" aria-describedby="fig4-description" width="600" height="363">
+   </a>
+   <div id="fig4-description" class="visually-hidden">Diagramme à barres montrant 3 appareils différents&nbsp;: en haut, un Pixel 3 présentant une petite portion à la fois sur le fil principal et le fil secondaire de moins de 400&nbsp;ms. Dans le cas du Moto G4, cela représente environ 900&nbsp;ms sur le fil principal et 300&nbsp;ms sur le fil secondaire. La dernière barre est celle d’un Alcatel 1X 5059D avec plus de 2&#8239;000&nbsp;ms sur le fil principal et plus de 500&nbsp;ms sur le fil secondaire.</div>
+   <figcaption id="fig4-caption">Figure 4. Délais de traitement de JavaScript pour Reddit.com, issus de <a href="https://v8.dev/blog/cost-of-javascript-2019">The cost of JavaScript in 2019 (en anglais)</a>.</figcaption>
```
Comment (bazzadp) [inline src/content/fr/2019/javascript.md:75]: Add a `lang="en"`?:

```suggestion
   <figcaption id="fig4-caption">Figure 4. Délais de traitement de JavaScript pour Reddit.com, issus de <a href="https://v8.dev/blog/cost-of-javascript-2019"><i lang="en">The cost of JavaScript in 2019</i> (en anglais)</a>.</figcaption>
```

Note can't put it on the `<a href...>` element as not all of it is in English.

File: src/content/fr/2019/javascript.md [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+
+<figure>
+   <a href="/static/images/2019/javascript/fig2.png">
+      <img src="/static/images/2019/javascript/fig2.png" alt="Figure 2. Distribution de JavaScript par page et par type d’appareil." aria-labelledby="fig2-caption" aria-describedby="fig2-description" width="600" height="371" data-width="600" data-height="371" data-seamless data-frameborder="0" data-scrolling="no" data-iframe="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpzDb9HGbdVvin6YPTOmw11qBVGGysltxmH545fUfnqIThAq878F_b-KxUo65IuXaeFVSnlmJ5K1Dm/pubchart?oid=1914565673&format=interactive">
+   </a>
+   <div id="fig2-description" class="visually-hidden">Un diagramme à barres montrant que 76 octets&#8239;/&#8239;65 octets de JavaScript sont utilisés dans le percentile p10 sur ordinateur de bureau et mobile respectivement, 186&#8239;/&#8239;164 octets pour p25, 391&#8239;/&#8239;359 octets pour p50, 721&#8239;/&#8239;668 octets pour p75, et 1&#8239;/&#8239;131&#8239;/&#8239;1&#8239;/&#8239;060 octets pour p90.</div>
```
Comment (bazzadp) [inline src/content/fr/2019/javascript.md:46]: Is it a bit OTT to use `&#8239` so much?

**Comment by bazzadp** [comment]
Actually @borisschapira could you translate the JavaScript details in `featured_chapters.html` and also enable `javascript` in line 1 of that file?

**Comment by borisschapira** [comment]
Yes, I'll do that ASAP.

File: src/content/fr/2019/javascript.md (hunk 2 above)
Comment (borisschapira) [reply src/content/fr/2019/javascript.md:46]: Is it ?

File: src/content/fr/2019/javascript.md [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+   </a>
+   <div id="fig9-description" class="visually-hidden">Diagramme à barres montrant 0&#8239;/&#8239;17 octets de JavaScript téléchargés sur mobiles issus de requêtes vers le domaine principal et, respectivement, vers des domaines tiers, dans le percentile p10, 6&#8239;/&#8239;54 en p25, 83&#8239;/&#8239;217 en p50, 189&#8239;/&#8239;477 en p75, et 380&#8239;/&#8239;827 en p90.</div>
+   <figcaption id="fig9-caption">Figure 9. Répartition de l’ensemble du JavaScript téléchargé sur mobile.</figcaption>
+</figure>
+
+À la médiane, on utilise 89&nbsp;% plus de code tiers que de code du domaine principal rédigé par l’équipe de développement, tant sur mobiles que sur ordinateurs de bureau. Cela montre clairement que le code tiers peut être un des principaux facteurs de lourdeur d’une page web. Pour plus d’informations sur l’impact des tiers, consultez le chapitre ["Tierces Parties"](./third-parties).
```
[the remark this answers is not in the document]
Comment (borisschapira) [reply src/content/fr/2019/javascript.md:132]: En effet, corrigé pour "Compression" aussi.

File: src/content/fr/2019/javascript.md [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+
+## Compression des ressources
+
+Dans le contexte des interactions entre navigateur et serveur, la compression des ressources fait référence au code qui a été transformé à l’aide d’un algorithme de compression des données. Les ressources peuvent être compressées statiquement à l’avance ou à la volée, quand le navigateur les requêtes. Dans les deux cas, la taille de la ressource transférée est considérablement réduite, ce qui améliore les performances de la page.
+
+Il existe de nombreux algorithmes de compression de texte, mais seuls deux sont principalement utilisés pour la compression (et la décompression) des requêtes sur le réseau HTTP&nbsp;:
+
+- [Gzip](https://www.gzip.org/) (gzip): le format de compression le plus utilisé pour les interactions entre serveurs et clients&nbsp;;
```
[the remark this answers is not in the document]
Comment (borisschapira) [reply src/content/fr/2019/javascript.md:140]: Bien vu, j'ai vu le deuxième.

File: src/content/fr/2019/javascript.md [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+- [Brotli](https://github.com/google/brotli) (br): un algorithme de compression plus récent visant à améliorer encore les taux de compression. [90&nbsp;% des navigateurs](https://caniuse.com/#feat=brotli) supportent la compression Brotli.
+
+Les scripts compressés devront toujours être décompressés par le navigateur une fois transférés. Cela signifie que son contenu reste le même et que les temps d’exécution ne sont pas du tout optimisés. Cependant, la compression des ressources améliorera toujours les temps de téléchargement, qui est également l’une des étapes les plus coûteuses du traitement JavaScript. S’assurer que les fichiers JavaScript sont correctement compressés peut constituer un des principaux facteurs d’amélioration des performances pour un site web.
```
[the remark this answers is not in the document]
Comment (borisschapira) [reply src/content/fr/2019/javascript.md:143]: Hum bien vu. Je vais opter pour "Cependant, la compression des ressources améliorera toujours leur temps de téléchargement, qui est également l’une des étapes les plus coûteuses du traitement JavaScript.", qui apporte la modification sur "les temps" afin de basculer au singulier.

File: src/content/fr/2019/javascript.md [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+
+Au cours des dernières années, l’écosystème JavaScript a connu une augmentation du nombre de librairies et <span lang="en">frameworks</span> open-source pour faciliter la création d’applications monopages (SPA). Une application monopage se caractérise par une page web qui charge une seule page HTML et utilise JavaScript pour modifier la page lors de l’interaction avec l’utilisateur au lieu de télécharger de nouvelles pages depuis le serveur. Bien que cela reste la principale prémisse des applications monopages, différentes approches de rendu du serveur peuvent encore être utilisées pour améliorer l’expérience de ces sites. Combien de sites utilisent ces types de <span lang="en">frameworks</span>&nbsp;?
```
[the remark this answers is not in the document]
Comment (borisschapira) [reply src/content/fr/2019/javascript.md:296]: Remplacé partout.

File: src/content/fr/2019/javascript.md [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+   </a>
+   <div id="fig13-description" class="visually-hidden">Diagramme à barres montrant que 0,6&nbsp;% des sites sur ordinateurs de bureau utilisent "type=module", et 0,8&nbsp;% des sites sur mobiles.</div>
+   <figcaption id="fig13-caption">Figure 13. Pourcentage de sites utilisant type=module.</figcaption>
+</figure>
+
+La prise en charge des modules au niveau du navigateur est encore relativement récente, et les chiffres montrent que très peu de sites utilisent actuellement le `type="module"` pour leurs scripts. De nombreux sites s’appuient encore sur des chargeurs de modules (2,37&nbsp;% de tous les sites sur ordinateurs de bureau utilisent [RequireJS](https://github.com/requirejs/requirejs) par exemple) et des bundlers ([webpack](https://webpack.js.org/) par exemple) pour définir des modules au sein de leur base de code.
```
[the remark this answers is not in the document]
Comment (borisschapira) [reply src/content/fr/2019/javascript.md:333]: Idem pour "bundles", c'est réglé, merci !

**Comment by borisschapira** [comment]
Thanks a lot @nico3333fr, all your remarks were relevant and have been taken into account. I knew I could count on you, it's so nice of you to do all this work right when you get back. I'll owe you one.

**Comment by bazzadp** [comment]
Thanks both. Merging!

**Pull request merged**